planner/core/rule_eliminate_projection.go

```diff
@@ -130,6 +131,12 @@ func doPhysicalProjectionElimination(p PhysicalPlan) PhysicalPlan {
 // eliminatePhysicalProjection should be called after physical optimization to
 // eliminate the redundant projection left after logical projection elimination.
 func eliminatePhysicalProjection(p PhysicalPlan) PhysicalPlan {
+	failpoint.Inject("DisableEliminatingPhysicalProjection", func(val failpoint.Value) {
```

Comment:
After eliminating some projections, the corresponding plan cost was not updated, which can lead our new and old cost implementation to produce different costs.